game-core/src/main/java/games/strategy/engine/framework/ui/SaveGameFileChooser.java

```diff
@@ -50,7 +50,17 @@
 
     AUTOSAVE_ODD(getOddRoundAutoSaveFile(true)),
 
-    AUTOSAVE_EVEN(getEvenRoundAutoSaveFile(true));
+    AUTOSAVE_EVEN(getEvenRoundAutoSaveFile(true)),
+
+    AUTOSAVE_END_TURN(getAutoSaveFile(addExtension("autosaveBeforeEndTurn"), true)),
```

Comment:
Could you re-use the `getBeforeStepAutoSaveFile()` method here (and other for other "before" steps)?